### ISSUE TITLE
Unable to pin resources via ipfs-companion

### ISSUE THREAD

**Issue by swedneck** [ipfs-shipyard/ipfs-companion#542]
Error while pinning IPFS Resource
JSON.parse: unexpected character at line 1 column 1 of the JSON data

**Comment by lidel** [comment]
Are you able to tell what was the raw JSON payload that triggered this error? 
Or what was the URL when you tried to pin?

Opening Browser Console (ctrl+shift+j) and enabling XHR and Request logging should help:
![screenshot_13](https://user-images.githubusercontent.com/157609/43260767-ac4f9604-90da-11e8-8ec9-21b4c97b0a16.png)

**Comment by swedneck** [comment]
IPFS companion Beta @ 9424acc
go-ipfs 0.4.16

I get the error no matter what i try to pin.

**Comment by lidel** [comment]
Hm.. I was unable to reproduce with these versions.

What OS are you running?  Is it possible some sort of firewall software is blocking request ?
Does below command succeeds and returns `{"Pins":["zb2rhk6GMPQF3hfzwXTaNYFLKomMeC6UXdUt6jZKPpeVirLtV"]}` ?

    curl 'http://127.0.0.1:5001/api/v0/pin/add?r=true&arg=bafkreigh2akiscaildcqabsyg3dfr6chu3fgpregiymsck7e7aqa4s52zy'

**Comment by swedneck** [comment]
I'm running fedora workstation 28, and that command returns the correct string.

**Comment by lidel** [comment]
@swedneck  and what is the browser version?

**Comment by swedneck** [comment]
Firefox 61.0.1

**Comment by lidel** [comment]
Hm.. are you running a lot of extensions in Firefox? 
What if you install Companion in an empty, ephemeral profile:

```bash
firefox --no-remote --profile $(mktemp -d) 'https://addons.mozilla.org/en-US/firefox/addon/ipfs-companion/'
```
If you try to pin via Companion running in clean Firefox, is the error still there?

**Comment by swedneck** [comment]
Nope, i get the same error.

**Comment by lidel** [comment]
So I've been running out of ideas :sweat_smile: and tried to pin/unpin IPNS resource, and bingo:
![screenshot_15](https://user-images.githubusercontent.com/157609/43320901-792473b4-91aa-11e8-9da2-16c097e8316d.png)

The error is caused by change in js-ipfs-api: in past `ipfs.name.resolve` returned object with `Path` field, right now it returns a path as a string. 

Will fix this shortly.

**Comment by swedneck** [comment]
Pinning that worked!

**Comment by lidel** [comment]
Fix landed in [v2.4.3.10570](https://github.com/ipfs-shipyard/ipfs-companion/releases/tag/v2.4.3.10570), thanks for reporting!
